cilium-cli/features/status.go

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package features
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/cilium-cli/defaults"
+)
+
+// perNodeMetrics maps a node name to their metrics
+type perNodeMetrics map[string][]*models.Metric
+
+// PrintFeatureStatus prints encryption status from all/specific cilium agent pods.
+func (s *Feature) PrintFeatureStatus(ctx context.Context) error {
+	ctx, cancelFn := context.WithTimeout(ctx, s.params.WaitDuration)
+	defer cancelFn()
+
+	pods, err := s.fetchCiliumPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	nodeMap, err := s.fetchStatusConcurrently(ctx, pods)
+	if err != nil {
+		return err
+	}
+
+	w := os.Stdout
+	if s.params.Outputfile != "-" {
+		w, err = os.Create(s.params.Outputfile)
+		if err != nil {
+			return err
+		}
+		defer w.Close()
+	}
+
+	switch s.params.Output {
+	case "tab":
+		return printPerNodeFeatureStatus(nodeMap, newTabWriter(w))
+	case "markdown":
+		return printPerNodeFeatureStatus(nodeMap, newMarkdownWriter(w))
+	case "json":
+		return printPerNodeStatusJson(nodeMap, w)
+	default:
+		return fmt.Errorf("output %s not recognized", s.params.Output)
+	}
+}
+
+func (s *Feature) fetchStatusConcurrently(ctx context.Context, pods []corev1.Pod) (perNodeMetrics, error) {
+	// res contains data returned from cilium pod
+	type res struct {
+		nodeName string
+		status   []*models.Metric
+		err      error
+	}
+	resCh := make(chan res)
+	defer close(resCh)
+
+	// concurrently fetch state from each cilium pod
+	for _, pod := range pods {
+		go func(ctx context.Context, pod corev1.Pod) {
+			st, err := s.fetchCiliumFeatureMetricsFromPod(ctx, pod)
+			resCh <- res{
+				nodeName: pod.Spec.NodeName,
+				status:   st,
+				err:      err,
+			}
+		}(ctx, pod)
+	}
+
+	// read from the channel, on error, store error and continue to next node
+	var err error
+	data := make(perNodeMetrics)
+	for range pods {
```

Comment:
If something goes south with any mx collectors, it might be better to range over the channel instead